lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/util/LibertyUtils.java

```diff
@@ -104,6 +110,49 @@ public static Path findFileInWorkspace(String serverXmlURI, String filename) {
         }
     }
 
+    /**
+     * Given a server.xml, determine Liberty runtime type of the LibertyWorkspace.
+     * 
+     * Will return 'wlp', 'ol', or null depending on the relevant property files found in the wlp/libs/version folder.
+     * 
+     * @param serverXMLUri server xml uri
+     * @return Liberty runtime or null
+     */
+    public static String getRuntimeInfo(String serverXMLUri) {
+
+        // return runtime set in settings if it exists
+        String libertyRuntime = SettingsService.getInstance().getLibertyRuntime();
+        if (libertyRuntime != null) {
+            return libertyRuntime;
+        }
+        // find workspace folder this serverXML belongs to
+        LibertyWorkspace libertyWorkspace = LibertyProjectsManager.getInstance().getWorkspaceFolder(serverXMLUri);
+
+        if (libertyWorkspace == null || libertyWorkspace.getWorkspaceURI() == null) {
+            return null;
+        }
+
+        String runtime = libertyWorkspace.getLibertyRuntime();
+
+        // return version from cache if set and Liberty is installed
+        if (runtime != null && libertyWorkspace.isLibertyInstalled()) {
```

Comment:
If the Liberty installation gets updated (they change the version in their pom.xml), will this get updated?

Reply:
It generates a new xsd file with the correct version, but the old version isn't deleted. Confirmed that it is using the new version.